gwpy/timeseries/timeseries.py

```diff
@@ -1873,22 +1924,15 @@ def gate(self, tzero=1.0, tpad=0.5, whiten=True,
         --------
         TimeSeries.mask
             for the method that masks out unwanted data
+        TimeSeries.find_gates
+            for the method that identifies gating points
         TimeSeries.whiten
             for the whitening filter used to identify gating points
         """
-        from scipy.signal import find_peaks
-        # Find points to gate based on a threshold
-        sample = self.sample_rate.to('Hz').value
-        data = self.whiten(**whiten_kwargs) if whiten else self
-        window_samples = cluster_window * sample
-        gates = find_peaks(abs(data.value), height=threshold,
-                           distance=window_samples)[0]
-        # represent gates as time segments
-        deadtime = SegmentList([Segment(
-            self.t0.value + (k / sample) - tzero,
-            self.t0.value + (k / sample) + tzero,
-        ) for k in gates]).coalesce()
-        # return the self-gated timeseries
+        deadtime = self.find_gates(tzero=tzero, whiten=whiten,
+                                   threshold=threshold,
+                                   cluster_window=cluster_window,
+                                   **whiten_kwargs)
         return self.mask(deadtime=deadtime, const=0, tpad=tpad)
```

Comment:
The name `gates` I think would be clearer here.

```suggestion
        gates = self.find_gates(tzero=tzero, whiten=whiten,
                                   threshold=threshold,
                                   cluster_window=cluster_window,
                                   **whiten_kwargs)
        return self.mask(deadtime=gates, const=0, tpad=tpad)
```